test/Scrutor.Tests/ScanningTests.cs

```diff
@@ -140,7 +175,7 @@ public void AutoRegisterAsMatchingInterface()
                     .AsMatchingInterface()
                     .WithTransientLifetime());
 
-            Assert.Equal(2, Collection.Count);
+            Assert.Equal(3, Collection.Count);

```

Comment:
Nah. I think DeclaredInterfaces and self is enough. Again, people can annotate subclasses or be more explicit about service types if they want to. This is simply for convenience :smile: